src/main/java/net/dv8tion/jda/api/interactions/components/text/TextInput.java

```diff
@@ -0,0 +1,330 @@
+    }
+
+    /**
+     * Creates a new TextInput Builder.
+     *
+     * @param id The custom id
+     * @param label The label
+     * @param style The {@link TextInputStyle TextInputStyle}
+     *
+     * @return a new TextInput Builder.
+     */
+    static TextInput.Builder create(@NotNull String id, @NotNull String label, @NotNull TextInputStyle style)
+    {
+        Checks.notNull(id, "Custom ID");
+        Checks.notNull(label, "Label");
+        Checks.notNull(style, "Style");
+        return new Builder(id, label, style);
+    }
+
+    /**
+     * Builder for {@link TextInput TextInputs}
+     */
+    class Builder
+    {
+        private final String id;
+        private final TextInputStyle style;
+        private final String label;
+
+        private int minLength = -1;
+        private int maxLength = -1;
+        private String value;
+        private String placeholder;
+        private boolean required;
+
+        protected Builder(String id, String label, TextInputStyle style)
+        {
+            this.id = id;
+            this.label = label;
+            this.style = style;
+        }
+
+        /**
+         * Sets whether the user is required to write in this TextInput.
+         *
+         * @param required If this TextInput should be required
+         *
+         * @return Builder for chaining convenience
+         */
+        public Builder setRequired(boolean required)
+        {
+            this.required = required;
+            return this;
+        }
+
+        /**
+         * Sets the minimum length of this input field.
+         *
+         * <b>This has to be between 0 and 4000</b>
+         *
+         * @param minLength The minimum amount of characters that need to be written
+         *
+         * @throws IllegalArgumentException
+         *         If minLength is negative or greater than 4000
+         *
+         * @return Builder for chaining convenience
+         */
+        public Builder setMinLength(int minLength)
+        {
+            Checks.notNegative(minLength, "Minimum length");
+            Checks.check(minLength <= 4000, "Minimum length cannot be longer than 4000 characters!");
+            this.minLength = minLength;
+            return this;
+        }
+
+        /**
+         * Sets the maximum length of this input field.
+         *
+         * <b>This has to be between 1 and 4000</b>
+         *
+         * @param maxLength The maximum amount of characters that need to be written
+         *
+         * @throws IllegalArgumentException
+         *         If maxLength is smaller than 1 or greater than 4000
+         *
+         * @return Builder for chaining convenience
+         */
+        public Builder setMaxLength(int maxLength)
+        {
+            Checks.check(maxLength >= 1, "Maximum length cannot be smaller than 1");
+            Checks.check(maxLength <= 4000, "Maximum length cannot be longer than 4000 characters!");
```

Comment:
Create a `public static final int MAX_LENGTH_TEXT_INPUT = 4000` or a name like that, my naming is bad